File: RepoTransBench/target_projects/Java/C#/nzakas_props2js/public_tests/PropertyConverterPublicTest.cs
using System;
using Xunit;
using System.Collections.Generic;

namespace Props2Js.Tests.Public
{
    public class PropertyConverterPublicTest
    {
        [Fact]
        public void TestConvertToJson_withString_public()
        {
            var p = new Dictionary<string, string> { { "color", "blue" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""color"":""blue""", json);
        }

        [Fact]
        public void TestConvertToJson_withInt_public()
        {
            var p = new Dictionary<string, string> { { "answer", "42" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""answer"":42", json);
        }

        [Fact]
        public void TestConvertToJson_withFloat_public()
        {
            var p = new Dictionary<string, string> { { "ratio", "3.14" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""ratio"":3.14", json);
        }

        [Fact]
        public void TestConvertToJson_withBooleanTrue_public()
        {
            var p = new Dictionary<string, string> { { "active", "true" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""active"":true", json);
        }

        [Fact]
        public void TestConvertToJson_withBooleanFalse_public()
        {
            var p = new Dictionary<string, string> { { "deleted", "false" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""deleted"":false", json);
        }

        [Fact]
        public void TestConvertToJsonP_public()
        {
            var p = new Dictionary<string, string> { { "score", "77" } };
            string outVal = PropertyConverter.ConvertToJsonP(p, "pubcb");
            Assert.StartsWith("pubcb(", outVal);
            Assert.EndsWith(");", outVal.Trim());
        }

        [Fact]
        public void TestConvertToJavaScript_var_public()
        {
            var p = new Dictionary<string, string> { { "animal", "cat" } };
            string js = PropertyConverter.ConvertToJavaScript(p, "pubVar");
            Assert.StartsWith("var pubVar=", js);
            Assert.EndsWith(";", js.Trim());
        }

        [Fact]
        public void TestConvertToJavaScript_assignment_public()
        {
            var p = new Dictionary<string, string> { { "animal", "dog" } };
            string js = PropertyConverter.ConvertToJavaScript(p, "globals.pet");
            Assert.DoesNotContain("var ", js);
            Assert.StartsWith("globals.pet=", js);
            Assert.EndsWith(";", js.Trim());
        }

        [Fact]
        public void TestConvertToJson_withMixedTypes_public()
        {
            var p = new Dictionary<string, string>
            {
                { "note", "hello" },
                { "age", "30" },
                { "pi", "3.1416" },
                { "success", "true" },
                { "error", "false" }
            };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""note"":""hello""", json);
            Assert.Contains(@"""age"":30", json);
            Assert.Contains(@"""pi"":3.1416", json);
            Assert.Contains(@"""success"":true", json);
            Assert.Contains(@"""error"":false", json);
        }

        [Fact]
        public void TestConvertToJson_handlesEmptyProperties_public()
        {
            var p = new Dictionary<string, string>();
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Equal("{}", json);
        }
    }
}

File: RepoTransBench/source_projects/Java/nzakas_props2js/tests/net/nczonline/web/props2js/PropertyConverterPublicTest.java
package net.nczonline.web.props2js;

import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Properties;

public class PropertyConverterPublicTest {

    @Test
    public void testConvertToJson_withString_public() {
        Properties p = new Properties();
        p.setProperty("color", "blue");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"color\":\"blue\""));
    }

    @Test
    public void testConvertToJson_withInt_public() {
        Properties p = new Properties();
        p.setProperty("answer", "42");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"answer\":42"));
    }

    @Test
    public void testConvertToJson_withFloat_public() {
        Properties p = new Properties();
        p.setProperty("ratio", "3.14");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"ratio\":3.14"));
    }

    @Test
    public void testConvertToJson_withBooleanTrue_public() {
        Properties p = new Properties();
        p.setProperty("active", "true");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"active\":true"));
    }

    @Test
    public void testConvertToJson_withBooleanFalse_public() {
        Properties p = new Properties();
        p.setProperty("deleted", "false");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"deleted\":false"));
    }

    @Test
    public void testConvertToJsonP_public() {
        Properties p = new Properties();
        p.setProperty("score", "77");
        String out = PropertyConverter.convertToJsonP(p, "pubcb");
        assertTrue(out.startsWith("pubcb("));
        assertTrue(out.endsWith(");"));
    }

    @Test
    public void testConvertToJavaScript_var_public() {
        Properties p = new Properties();
        p.setProperty("animal", "cat");
        String js = PropertyConverter.convertToJavaScript(p, "pubVar");
        assertTrue(js.startsWith("var pubVar="));
        assertTrue(js.endsWith(";"));
    }

    @Test
    public void testConvertToJavaScript_assignment_public() {
        Properties p = new Properties();
        p.setProperty("animal", "dog");
        String js = PropertyConverter.convertToJavaScript(p, "globals.pet");
        assertFalse(js.startsWith("var "));
        assertTrue(js.startsWith("globals.pet="));
        assertTrue(js.endsWith(";"));
    }

    @Test
    public void testConvertToJson_withMixedTypes_public() {
        Properties p = new Properties();
        p.setProperty("note", "hello");
        p.setProperty("age", "30");
        p.setProperty("pi", "3.1416");
        p.setProperty("success", "true");
        p.setProperty("error", "false");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"note\":\"hello\""));
        assertTrue(json.contains("\"age\":30"));
        assertTrue(json.contains("\"pi\":3.1416"));
        assertTrue(json.contains("\"success\":true"));
        assertTrue(json.contains("\"error\":false"));
    }

    @Test
    public void testConvertToJson_handlesEmptyProperties_public() {
        Properties p = new Properties();
        String json = PropertyConverter.convertToJson(p);
        assertEquals("{}", json);
    }
}